lib/cli.js

```diff
@@ -21,82 +21,70 @@ var fs = require('fs');
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = gitDirectory + '/logs/incorrect-commit-msgs';
+  }
+} catch (err) {}
+
+validate(commitMsgFileOrText);
+
+
+function bufferToString(buffer) {
```

Comment:
How positive is it that there are no side effects by redefining this even in mocha runs? The original code seemed to guard this being added to only when it was needed. I'm wary of removing that guard without knowing it is safe.